### DISCUSSION TITLE
Questions about `SlotProps.slot` in `Slots.tsx` implementation

### DISCUSSION THREAD

**Discussion by bstro** [adobe/react-spectrum#3449] [Q&A] [answered]
Hi there! I'm trying to understand exactly how `Slots.tsx` works. In particular, I'm wondering about [useSlotProps](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/utils/src/Slots.tsx#L22):

```
interface SlotProps {
  slot?: string
}

export function useSlotProps<T>(props: T & {id?: string}, defaultSlot?: string): T {
  let slot = (props as SlotProps).slot || defaultSlot;
  let {[slot]: slotProps = {}} = useContext(SlotContext) || {};

  return mergeProps(props, mergeProps(slotProps, {id: props.id}));
}
```

What is `SlotProps.slot` intended for? Is it so that I could declare the child's slot from the parent?

In all the invocations of `useSlotProps` that I've seen so far in the Spectrum codebase, they all seem to specify a defaultProp. All except one: `View.tsx`'s invocation of `useSlotProps` has no second parameter for `defaultSlot`. I am wondering why this is the case.

Thank you!

**Comment by LFDanLu** [comment, marked as answer]
> What is SlotProps.slot intended for? Is it so that I could declare the child's slot from the parent?

Yes that is correct, here is an example of [this](https://github.com/adobe/react-spectrum/blob/13696a97a7742eb84cb31f5cb50123a2da8f090e/packages/%40react-spectrum/listbox/stories/ListBox.stories.tsx#L500-L504). As you can see in this story, there are two `Text` elements, one without a `slot` prop defined (and thus has the default "text" slot) and one with `slot="description"`. This allows us to send the [proper classNames/props to each](https://github.com/adobe/react-spectrum/blob/13696a97a7742eb84cb31f5cb50123a2da8f090e/packages/%40react-spectrum/listbox/src/ListBoxOption.tsx#L101-L106) so that one renders as the main ListBox's option's text and the other as the option's description:
<img width="251" alt="image" src="https://user-images.githubusercontent.com/9661127/186720519-450b6720-bc4e-41cc-ac6c-4a192227dadb.png">

View doesn't have a default slot since it is a more generic container rather than a commonly used component that is slotted within one of our components.

**Reply by bstro** [reply]
this makes sense! thank you!
